types/params.go

```diff
@@ -81,18 +85,23 @@ func (params *ValidatorParams) IsValidPubkeyType(pubkeyType string) bool {
 // Validate validates the ConsensusParams to ensure all values are within their
 // allowed limits, and returns an error if they are not.
 func (params *ConsensusParams) Validate() error {
-	if params.BlockSize.MaxBytes <= 0 {
-		return cmn.NewError("BlockSize.MaxBytes must be greater than 0. Got %d",
-			params.BlockSize.MaxBytes)
+	if params.Block.MaxBytes <= 0 {
+		return cmn.NewError("Block.MaxBytes must be greater than 0. Got %d",
+			params.Block.MaxBytes)
 	}
-	if params.BlockSize.MaxBytes > MaxBlockSizeBytes {
-		return cmn.NewError("BlockSize.MaxBytes is too big. %d > %d",
-			params.BlockSize.MaxBytes, MaxBlockSizeBytes)
+	if params.Block.MaxBytes > MaxBlockSizeBytes {
+		return cmn.NewError("Block.MaxBytes is too big. %d > %d",
+			params.Block.MaxBytes, MaxBlockSizeBytes)
 	}
 
-	if params.BlockSize.MaxGas < -1 {
-		return cmn.NewError("BlockSize.MaxGas must be greater or equal to -1. Got %d",
-			params.BlockSize.MaxGas)
+	if params.Block.MaxGas < -1 {
+		return cmn.NewError("Block.MaxGas must be greater or equal to -1. Got %d",
+			params.Block.MaxGas)
+	}
+
+	if params.Block.TimeIotaMs < 0 {
```

Comment:
Can this really be 0?

Reply:
It should probably be always positive so we ensure time monotonicity.